src/diff/changes/apiChanges.ts

```diff
@@ -76,4 +98,48 @@ export class ApiChanges {
   getIgnoredChangesCount(): number {
     return this.getChangeCountByCategory(RuleCategory.IGNORED);
   }
+
+  /**
+   * Gets the number of changes in each category
+   */
+  getCategorySummary(): CategorySummary {
+    return this.nodeChanges.reduce((summary, node) => {
+      const nodeSummary = node.getCategorySummary();
+      Object.entries(nodeSummary).forEach(([category, count]) => {
+        summary[category] += count;
+      });
+      return summary;
+    }, createCategorySummary());
+  }
+
+  /**
+   * Returns data in a format more easily consumed by Handlebars templates
+   */
+  getTemplateData(): ApiChangesTemplateData {
+    const nodeChanges = this.getNodesWithCategorizedChanges();
+    return {
+      nodeChanges: nodeChanges.map((nc) => nc.getTemplateData()),
+      apiSummary: this.getCategorySummary(),
+    };
+  }
+
+  /**
+   * Render the changes as a formatted string
+   * @param format Name of format to use
+   * @see DiffCommand for a list of supported formats
+   */
+  toFormattedString(format: string): string {
```

Comment:
overall looks good compared to the previous logic. I still think data should be separated from formatter mainly for the extendibility reasons. Probably fine for now